src/Planner/PlannerExpressionAnalysis.cpp

```diff
@@ -467,6 +479,12 @@ SortAnalysisResult analyzeSort(const QueryNode & query_node,
                 input_node_it = it;
             }
 
+            if (aggregation_analysis_result_optional)
+                if (!aggregation_keys.contains(node.result_name))
+                    throw Exception(ErrorCodes::NOT_AN_AGGREGATE,
+                        "Column {} is not under aggregate function and not in GROUP BY keys. In query {}",
```

Comment:
This check is done in QueryAnalysisPass. Why do you add it here>

Reply:
consider next select:
```
SELECT n, number+5 AS inter FROM (
   SELECT toFloat32(number % 10) AS n, number, number*2 AS mn
   FROM numbers(10) WHERE number % 3 = 1
) GROUP BY n, inter ORDER BY n WITH FILL FROM 0 TO 5.51 STEP 0.5 INTERPOLATE (inter AS mn * 2);
```
column `mn` is required for calculation, however it's not under any aggregation and will be missing ending up with exception (when without this interception):
```
Code: 10. DB::Exception: Not found column __table1.mn: in block __table1.n Float32 Float32(size = 0), plus(__table1.number, 5_UInt8) UInt64 UInt64(size = 0). (NOT_FOUND_COLUMN_IN_BLOCK)
```
here:
https://github.com/ClickHouse/ClickHouse/blob/fix-analyzer-interpolate/src/Interpreters/ActionsDAG.cpp#L824

this exception while correct, yet absolutely not clear - so we intercept this situation early and generate correct exception:
```
Code: 215. DB::Exception: Column __table1.mn is not under aggregate function and not in GROUP BY keys. In query SELECT n, number + 5 AS inter FROM (SELECT toFloat32(number % 10) AS n, number, number * 2 AS mn FROM numbers(10) WHERE
 (number % 3) = 1) GROUP BY n, inter ORDER BY n ASC WITH FILL FROM 0 TO 5.51 STEP 0.5 INTERPOLATE ( inter AS mn * 2 ). (NOT_AN_AGGREGATE)
```

Reply:
Thanks for the explanation. Now I understand the error, but I still think the fix is wrong. The issue is located here:
https://github.com/ClickHouse/ClickHouse/blob/9fb8c8263f85322f3ae3b190a60110226a6f9e56/src/Analyzer/ValidationUtils.cpp#L83-L84

Validation visitor should also check INTERPOLATE expressions.